src/app/core/models/templates/template.model.ts

```diff
@@ -23,7 +23,12 @@ export abstract class Template {
    * Check whether the given string conforms to the template.
    */
   test(toTest: string): boolean {
-    return this.regex.test(toTest);
+    let numOfMatch = 0;
+    while (this.regex.exec(toTest) != null) {
+      numOfMatch += 1;
+    }
+    this.regex.lastIndex = 0;
+    return numOfMatch === this.headers.length;
```

Comment:
``Also there was an issue with finding the right comment with the correct format to parse which I have also fixed.``

Reply:
Sorry, could I clarify what the existing problem was here?

Reply:
The resulting regex for team template is:
```
/(# Team's Response|## Duplicate status \(if any\):)\s+([\s\S]*?)(?=# Team's Response|## Duplicate status \(if any\):|$)/gi
```
This regex recognises the following as the correct template:
```
# Team's Response
## Duplicate status (if any):
--
```
The number of matched groups however is only 1. But we need 2 groups to be matched for the template to be correct.

Reply:
Ok, understood. I have another question: Currently, when we want to edit a team response, CATcher will record our new response (via the form), and locate the comment to be edited by finding the first comment in the thread that matches the template regex.

However, why doesn't CATcher throw any error (or warning) is there more than one comment matching the team response template?

Reply:
It would be good to have a visual cue to indicate that there are duplicated responses. However such a situation should not happen if they use CATcher, even more so after polling is implemented. Unless we they make changes via Github.

Reply:
Nonetheless, it is still possible for users to post duplicated new response if they submit at the exact same time.